### PR TITLE
Implement Debian packaging

### PR DESCRIPTION
**Pull request by ottok**
Based on existing official Debian packaging at
https://salsa.debian.org/python-team/applications/rdiff-backup

### CODE REVIEW

**Comment by ericzolf** [comment]
I'll definitely have a look but one question ahead: from my past as Debian packager, I remember that "debian" folders upstream were rather disturbing than helpful; don't you fear issues when you'll package for Debian "downstream"?

**Comment by ericzolf** [comment]
I remember now what the issue was: with the `/debian/` directory in the way, you need to remove it from the sources before you re-add it from somewhere else, or your package looks "native", which only Debian-related packages are allowed to be.

My suggestion:

1. move `debian` to `tools` directory
2. ignore `/debian/` in `.gitignore` (the slashes are important)
3. add a step to copy the debian directory back into the root directory before building

Else, random thought (still struggling to get the build running without root access using podman): it would be good to have the .deb land into the `dist/` directory with all the other packages.

**Comment by ottok** [comment]
I can automatically filter away `/debian/` when importing the tar.gz into Debian if I want to, this is not a problem. On the contrary it will be beneficial that there is a properly maintained `debian/` in upstream where changes go hand-in-hand with real needs as the package evolves.

**Comment by ottok** [comment]
Related: https://github.com/rdiff-backup/rdiff-backup/issues/55#issuecomment-570087976

File: debian/copyright [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: rdiff-backup
+Upstream-Contact: rdiff-backup-users@nongnu.org
+Source: https://github.com/rdiff-backup/rdiff-backup
+
+Files: *
+Copyright: 2001-2008 Ben Escoto <bescoto@stanford.edu>
+           2019-2020 Eric Lavarde <eric@lavar.de>
```
Comment (ericzolf) [inline debian/copyright]: can you please use my rdiffbackup address? ewl+rdiffbackup-at-lavar.de

File: .travis.yml [hunk 2]
```diff
@@ -132,6 +132,28 @@ matrix:
      script:
        - docker run --rm -e PLAT=$PLAT -v `pwd`:/io $DOCKER_IMAGE $PRE_CMD /io/tools/build_wheels.sh
        - ls wheelhouse/
+   - os: linux
+     language: sh
+     env: MAKE_STEP=build_deb RUN_COMMAND=
```
Comment (ericzolf) [inline .travis.yml]: This sets RUN_COMMAND to empty and avoids running the gbp command in Docker during the Travis build. If it is what you wanted, fine as it works, just unsure.

File: Makefile [hunk 3]
```diff
@@ -46,6 +46,9 @@ bdist_wheel:
 	${RUN_COMMAND} ./setup.py bdist_wheel
 	# ${RUN_COMMAND} auditwheel repair dist/*.whl
 
+build_deb:
+	${RUN_COMMAND} gbp buildpackage -us -uc
```
Comment (ericzolf) [inline Makefile]: Can you please make sure that the .deb package lands under `dist/`? It makes it then easy to deploy the package with all other packages.

Comment (ottok) [reply Makefile]: It lands in the directory above the git source code. I am not aware of any option to tell debbuild to output in a selected directory. I'll check.

File: .travis.yml (hunk 2 above)
Comment (ottok) [reply .travis.yml]: Yes, this is intentional. The deb build does not benefit from any (extra) layer of Docker

File: debian/copyright (hunk 1 above)
Comment (ottok) [reply debian/copyright]: ok

File: Makefile (hunk 3 above)
Comment (ericzolf) [reply Makefile]: I was actually simply thinking of copying/moving the generated file to the right place.

Comment (ottok) [reply Makefile]: OK, this is now done in travis. This should be good enough to get merged now. The makefile and travis can be tweaked more later.

**Comment by ikus060** [comment]
This change will conflict with #231
Any particular reason to change all the formatting of the travis file ??

**Comment by ottok** [comment]
> This change will conflict with #231
> Any particular reason to change all the formatting of the travis file ??

As stated in the commit message, I fixed the indentation. If you check the original file in a YAML validator, it would fail. You can merge either one of the PRs and then the latter one can be rabased on master to fix the merge conflict.

File: .travis.yml [hunk 4]
```diff
@@ -17,121 +17,140 @@ deploy:
...
           all_branches: true
    # Build and deploy Linux wheels using manylinux Docker containers
    # avoiding manylinux2014_i686 because it does not provide librsync-devel
-   - os: linux
-     sudo: required
-     services:
-       - docker
-     env: DOCKER_IMAGE=quay.io/pypa/manylinux1_x86_64
+    - os: linux
+      services:
```
Comment (ericzolf) [inline .travis.yml]: Why was the `sudo: required` removed?

**Comment by ottok** [comment]
Rebased on latest master now.

**Comment by ottok** [comment]
Made a separate PR of the .travis-ci.yml future proofing fixes so this PR stays as a cleaner on the .deb topic.

**Comment by ericzolf** [comment]
I created the branch ericzolf-feature-deb-builds-dodeploy from your branch before the format clean-up and Travis didn't deploy any deb file. Then I realised that the deploy section is missing, can you please add one?

**Comment by ottok** [comment]
Can we just merge this now please and work on additional enhancements in future branches?
Having the debs in the deploy probably not useful at all and certainly not critical.

**Comment by ericzolf** [comment]
I'm still a bit questioning the point of having the Debian stuff in the travis CI if it's not to get it deployed, but I can merge for now. Let's continue the discussion on #221 and the overall Linux deployment.

This said, the Travis CI pipeline seems to be stuck and I can't merge. Can you amend the last commit and force push it to get a new try?

**Comment by ikus060** [comment]
> I'm still a bit questioning the point of having the Debian stuff in the travis CI

It's nice to know "when" we gonna break the Debian packaging. It's like another testing of the code. Check if the debian package is working.
But since it's so easy to deploy the deb on github, I would also add it.

**Comment by ericzolf** [comment]
OK, I use the last merge to refresh the branch and (hopefully) re-trigger the pipeline. Stay tuned...

**Comment by ericzolf** [comment]
Something is wrong in the Travis yaml: https://travis-ci.org/rdiff-backup/rdiff-backup/requests shows "Could not parse rdiff-backup/rdiff-backup:.travis.yml" - first time 19 hours ago...

**Comment by ottok** [comment]
I am looking into the Travis-CI issue now.

My local branch and this PR had diverged, maybe I screwed up and forgot the push the last rebase...

**Pull request merged**